includes/data-stores/class-wc-admin-reports-orders-data-store.php

```diff
@@ -207,6 +207,11 @@ public function get_data( $query_args ) {
 				$this->include_extended_info( $orders_data, $query_args );
 			}
 
+			foreach ( $orders_data as $key => $order_data ) {
+				$wc_order                            = new WC_Order( $order_data['order_id'] );
+				$orders_data[ $key ]['order_number'] = $wc_order->get_order_number();
```

Comment:
Any performance concerns here / or is there a way to determine if a site is using something that manipulates order numbers?

Reply:
Good question. There will definitely be extra queries from this, but I can't think of anyway else to get the order number. The `woocommerce_order_number` filter passes both the order ID and order object to plugins, so even if we reimplemented the filter, we would still need to load the object. Order data does get cached https://github.com/woocommerce/woocommerce/blob/de642211d08a81fb32addefa69cd1d0c1b6f7643/includes/abstracts/abstract-wc-order.php#L64.

However, this was a good suggestion for adding a check. We can use `has_filter` to avoid doing anything and just returning the order ID.